server/controllers/createLocation/confirm.ts

```diff
@@ -0,0 +1,99 @@
+import { NextFunction, Response } from 'express'
+import FormWizard from 'hmpo-form-wizard'
+import { capitalize } from 'lodash'
+import backUrl from '../../utils/backUrl'
+import { TypedLocals } from '../../@types/express'
+import FormInitialStep from '../base/formInitialStep'
+import { Location } from '../../data/types/locationsApi'
+import decorateLocation from '../../decorators/location'
+
+export default class ConfirmCreateLocation extends FormInitialStep {
+  middlewareSetup() {
+    super.middlewareSetup()
+  }
+
+  locals(req: FormWizard.Request, res: Response): Partial<TypedLocals> {
+    const locals = super.locals(req, res)
+    const { prisonId, locationId } = res.locals
+    const { locationType, structureLevels } = res.locals.values
+
+    // locations API uses singular types; UI needs to display them as plural.
+    const singularToPluralMap: Record<string, string> = {
+      LANDING: 'Landings',
+      CELL: 'Cells',
+      SPUR: 'Spurs',
+    }
+
+    const pluralize = (level: string) => singularToPluralMap[level] || capitalize(level.toLowerCase())
+
+    const fullStructure = [locationType, ...structureLevels]
+    locals.decoratedLocationStructure = fullStructure
+      .map((level, i) => (i === 0 ? capitalize(level) : pluralize(level)))
+      .join(' → ')
+
+    locals.backLink = backUrl(req, {
+      fallbackUrl: `/create-new/${[prisonId, locationId].filter(i => i).join('/')}/structure`,
+    })
+    locals.cancelLink = `/view-and-update-locations/${[prisonId, locationId].filter(i => i).join('/')}`
+    locals.createStructureLink = `/create-new/${[prisonId, locationId].filter(i => i).join('/')}/structure`
+    locals.createDetailsLink = `/create-new/${[prisonId, locationId].filter(i => i).join('/')}/details`
```

Comment:
`locationId || prisonId` is correct here, because the route is `/create-new/:prisonOrLocationId`